packages/astro/src/core/errors/errors-data.ts

```diff
@@ -976,7 +976,8 @@ export const RedirectWithNoLocation = {
 export const UnsupportedExternalRedirect = {
 	name: 'UnsupportedExternalRedirect',
 	title: 'Unsupported or malformed URL.',
-	message: 'An external redirect must start with http or https, and must be a valid URL.',
+	message: (from: string, to: string) => `The external redirect from "${from}" to "${to}" is invalid.`,
```

Comment:
Here are some ideas for you! See if you like anything here!

My motivation is: people don't like being told their stuff is "invalid" if it's just malformed for our system 😄 

```suggestion
	message: (from: string, to: string) => `The external redirect from "${from}" to "${to}" cannot be processed because the destination URL is unsupported.`,
```
OR, I kind of like focusing on the place you're redirecting to (since that is going to be the problem with this error message, not the original URL), so we don't HAVE to think of it as FROM (not a problem) going to TO (where the problem actually is) when we want to draw attention to the later thing...

```suggestion
	message: (from: string, to: string) => `The destination URL in the external redirect from "${from}" to "${to}" is unsupported.`,
```

Since the hint describes *why*, maybe this can just say, "we can't do it".

I'll leave it to you to choose, but just telling people their stuff is "invalid" when we could imply "not formatted properly" can feel a little harsh. 😄